tests/clock/test_clock.py

```diff
@@ -0,0 +1,427 @@
+
+    # expected errors
+    ERR_BAD_TIMEZONE = 'Timezone {} does not conform format'
+    ERR_MISSING_DATE = 'Error: Missing argument "<YYYY-MM-DD>"'
+    ERR_MISSING_TIME = 'Error: Missing argument "<HH:MM:SS>"'
+    ERR_BAD_DATE = 'Date {} does not conform format YYYY-MM-DD'
+    ERR_BAD_TIME = 'Time {} does not conform format HH:MM:SS'
+
+    # timedatectl
+    CMD_TIMEDATECTL = "timedatectl"
+    TIME_ZONE = "Time zone"
+
+    MIN_SYSTEM_DATE = "1970-01-01"
+    MAX_SYSTEM_DATE = "2231-12-31"
+
+    # ntp
+    CMD_SHOW_NTP = "show ntp"
+    CMD_CONFIG_NTP_ADD = "config ntp add"
+    CMD_CONFIG_NTP_DEL = "config ntp del"
+    OUTPUT_CMD_NTP_ADD_SUCCESS = 'NTP server {} added to configuration\nRestarting ntp-config service...'
+    OUTPUT_CMD_NTP_DEL_SUCCESS = 'NTP server {} removed from configuration\nRestarting ntp-config service...'
+    REGEX_NTP_POLLING_TIME = r'polling server every (\d+)'
+
+
+class ClockUtils:
+    @staticmethod
+    def run_cmd(duthosts, cmd, param=''):
+        """
+        @summary:
+            Run a given command and return its output.
+            * A successful command returns an empty output (except for show commands),
+                while failure returns an error message
+        @return: commands output (str)
+        """
+        with allure_step('Run command: "{}" with param "{}"'.format(cmd, param)):
+            dut_hostname = duthosts[0].hostname
+
+            cmd_to_run = cmd if param == '' else cmd + ' ' + param
+            logging.info('Actual command to run: "{}"'.format(cmd_to_run))
+
+            try:
+                cmd_output = duthosts.command(cmd_to_run)[dut_hostname][ClockConsts.STDOUT]
+            except RunAnsibleModuleFail as cmd_err:
+                output = cmd_err.results[ClockConsts.STDOUT]
+                err = cmd_err.results[ClockConsts.STDERR]
+                cmd_output = output if output else err
+                logging.info('Command Error!\nError message: "{}"'.format(cmd_output))
+
+            cmd_output = str(cmd_output)
+            logging.info('Output: {}'.format(cmd_output))
+
+        return cmd_output
+
+    @staticmethod
+    def verify_and_parse_show_clock_output(show_clock_output):
+        """
+        @summary:
+            Verify, and then split output of show clock into date, time and timezone strings
+
+            Exapmple:
+            "Mon 03 Apr 2023 11:29:46 PM UTC" -> {"date": "2023-04-03", "time": "23:29:46", "timezone": "+0000"}
+        @param show_clock_output: the given show clock output
+        @return: The splited output as a dict
+        """
+        with allure_step('Verify output of show clock'):
+            try:
+                timezone_str = show_clock_output.split()[-1].strip()
+                logging.info('Timezone str: "{}"'.format(timezone_str))
+
+                date_time_to_parse = show_clock_output.replace(timezone_str, '').strip()
+                logging.info('Time and date to parse: "{}"'.format(date_time_to_parse))
```

Comment:
if this script is running on Python3(which is the expectation), then f-string is more preferable that 'format'

Reply:
ok, working on it :-)